docs/writing_tilt_extensions.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Writing Tilt Extensions (Preview)
+layout: docs
+---
+
+Introducing Tilt Extensions! Extensions are a simple way to bundle up Tiltfile functionality that should be shared with the world.
```

Comment:
mentioned on slack, but as a non-blog-post, the first sentence here should probably feel less announce-y